File: generate-favicons.py
#!/usr/bin/env python3
from PIL import Image
import os

public_dir = os.path.join(os.path.dirname(__file__), 'public')
logo_path = os.path.join(public_dir, 'clavrr-logo.png')

try:
    # Open the logo
    img = Image.open(logo_path)
    
    # Create 256x256 favicon
    favicon = img.resize((256, 256), Image.Resampling.LANCZOS)
    favicon.save(os.path.join(public_dir, 'favicon.png'))
    print('✓ favicon.png created (256x256)')
    
    # Create 32x32
    favicon_32 = img.resize((32, 32), Image.Resampling.LANCZOS)
    favicon_32.save(os.path.join(public_dir, 'favicon-32x32.png'))
    print('✓ favicon-32x32.png created')
    
    # Create 16x16
    favicon_16 = img.resize((16, 16), Image.Resampling.LANCZOS)
    favicon_16.save(os.path.join(public_dir, 'favicon-16x16.png'))
    print('✓ favicon-16x16.png created')
    
    # Create apple-touch-icon (180x180)
    apple_icon = img.resize((180, 180), Image.Resampling.LANCZOS)
    apple_icon.save(os.path.join(public_dir, 'apple-touch-icon.png'))
    print('✓ apple-touch-icon.png created (180x180)')
    
except Exception as e:
    print(f'Error: {e}')

File: generate-favicons.js
const fs = require('fs');
const path = require('path');

// Using native Node.js with sharp if available, or fallback to simple file copy
try {
  const sharp = require('sharp');
  const sourceImg = path.join(__dirname, 'public', 'clavrr-logo.png');
  
  // Generate 256x256 favicon
  sharp(sourceImg)
    .resize(256, 256, { fit: 'contain', background: { r: 248, g: 245, b: 240, alpha: 1 } })
    .png()
    .toFile(path.join(__dirname, 'public', 'favicon.png'))
    .then(() => console.log('✓ favicon.png created (256x256)'))
    .catch(err => console.error('Error creating favicon.png:', err));

  // Generate 32x32 favicon.ico compatible PNG
  sharp(sourceImg)
    .resize(32, 32, { fit: 'contain', background: { r: 248, g: 245, b: 240, alpha: 1 } })
    .png()
    .toFile(path.join(__dirname, 'public', 'favicon-32x32.png'))
    .then(() => console.log('✓ favicon-32x32.png created'))
    .catch(err => console.error('Error creating favicon-32x32.png:', err));

  // Generate 16x16
  sharp(sourceImg)
    .resize(16, 16, { fit: 'contain', background: { r: 248, g: 245, b: 240, alpha: 1 } })
    .png()
    .toFile(path.join(__dirname, 'public', 'favicon-16x16.png'))
    .then(() => console.log('✓ favicon-16x16.png created'))
    .catch(err => console.error('Error creating favicon-16x16.png:', err));

} catch (err) {
  console.error('Sharp not available, please run: npm install sharp');
}
